### PR TITLE
fix #2845: fix panic on aggregate functions

### LINKED ISSUE #2845: Panic on query with aggregation functions in RC32
Except count() in RC32 queries with any aggregation functions raise exception.
An example query:
```
> select value from errors where if_id='3013975' and type='in_framing' and time > now() - 1h
name: errors
------------
time                    value
2015-06-09T07:42:47Z    2
2015-06-09T07:47:47Z    5
2015-06-09T07:52:49Z    1
2015-06-09T08:07:45Z    2
2015-06-09T08:27:46Z    6
2015-06-09T08:32:45Z    13
2015-06-09T08:37:46Z    5

>
> select mean(value) from errors where if_id='3013975' and type='in_framing' and time > now() - 1h
ERR: Get http://localhost:8086/query?db=observer&q=select+mean%28value%29+from+errors+where+if_id%3D%273013975%27+and+type%3D%27in_framing%27+and+time+%3E+now%28%29+-+1h+: EOF
```
An example of exception in influxdb.log

```
panic: interface conversion: interface is int64, not float64

goroutine 88 [running]:
github.com/influxdb/influxdb/influxql.MapMean(0x7fa6f9e19ed8, 0xc20f22e2c0, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/functions.go:455 +0xbf
github.com/influxdb/influxdb/tsdb.(*LocalMapper).NextInterval(0xc20f22e2c0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/tsdb/tx.go:378 +0xd3
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processAggregate(0xc214be7ce0, 0xc22a61af90, 0xb6cc20, 0xc22a5f6640, 0x1, 0x1, 0x0, 0x0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:778 +0x264
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc214be7ce0, 0xc2289062a0, 0xc20808f700)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:170 +0x8ca
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc20c699e40, 0xc2289062a0)
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:926 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
        /root/.gvm/pkgsets/go1.4.2/global/src/github.com/influxdb/influxdb/influxql/engine.go:905 +0x5a
```

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: influxql/functions.go [hunk 1]
```diff
@@ -415,16 +417,35 @@ func ReduceCountDistinct(values []interface{}) interface{} {
 	return len(index)
 }
 
+type NumberType int8
+
+const (
```
Comment (otoolep) [inline influxql/functions.go]: At first glance this seems to overlap with the `InspectDataType` in the same package. Not exactly the same, but it seems redundant.

**Comment by otoolep** [comment]
Not sure about this -- we already have the concepts of our own types in the `influxql` package. They are used by the field codecs. Should we not be using that concept here again?

**Comment by beckettsean** [comment]
May address #2704

**Comment by toddboom** [comment]
@dgnorton i'm getting this when i run the tests:
```
# github.com/influxdb/influxdb/influxql
influxql/functions_test.go:48: too few values in struct initializer
influxql/functions_test.go:55: too few values in struct initializer
```

that seems to be what's making the suite fail - are you seeing that locally?

**Comment by toddboom** [comment]
Taking a presumptive :+1: from @otoolep and my +1 to merge this in.

**Pull request merged**